### ISSUE TITLE
Support Tabbed Windows

### ISSUE THREAD

**Issue by ThomasAdam** [fvwmorg/fvwm3#472]
In fvwm2 there used to be a `FvwmTabs` module which would swallow windows into a tabbed interface.  Indeed, the program `tabbed` from suckless.org achieves same thing.

`Fvwm3` should support this natively instead:

* Windows should be `dragged` inside one another to support tabbing;
* The container titlebar should be split to indicate the number of tabbed windows;
* Dragging a titlebar out of a tabbed window should detach it;
* Style options for auto-tabbing;
* Key/mouse bindings to support this.

Although this feature could be a module, I think most paradigm shifts are such that most users would expect this feature "out-of-the-box" -- having this as a module is another layer of indirection to set up.

**Comment by lgsobalvarro** [comment]
Could the old FvwmTabs be used as a starting point? Or would it make things harder?

**Comment by danespen** [comment]
Never used tabs.
Seemed to me having the code in a module was a good thing
since it wasn't using storage if I didn't start it.

If a built in menu entry can be made to enable and configure it,
then it doesn't matter much how much has to be configured
to start it.


Thomas Adam ***@***.***> writes:

 In fvwm2 there used to be a FvwmTabs module which would swallow windows
 into a tabbed interface. Indeed, the program tabbed from suckless.org
 achieves same thing.
>
> Fvwm3 should support this natively instead:
>
> * Windows should be dragged inside one another to support tabbing; *
> The container titlebar should be split to indicate the number of
> tabbed windows; * Dragging a titlebar out of a tabbed window should
> detach it; * Style options for auto-tabbing; * Key/mouse bindings to
> support this.
>
> Although this feature could be a module, I think most paradigm shifts
> are such that most users would expect this feature "out-of-the-box" --
> having this as a module is another layer of indirection to set up.
>
> — You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
>

-- 
Dan Espen

**Comment by lgsobalvarro** [comment]
@danespen I agree. I never used FvwmTabs, not I see the need for them. Maybe keeping it as a module would be a better alternative. Possibly the original module could be cleaned up and updated in order to work well with FVWM3 instead. Hence why I was asking if the original module could be used as a starting point for what @ThomasAdam suggests.
